session_test.go

```diff
@@ -1973,20 +1967,16 @@ func (s *testSessionSuite) TestIssue1435(c *C) {
 	execFailedFunc := func(s Session, tbl string, start chan struct{}, end chan error) {
 		// execute successfully
 		_, err := exec(s, "begin;")
+		c.Check(err, IsNil)
 		<-start
 		<-start
-		if err == nil {
-			// execute failed
-			_, err = exec(s, fmt.Sprintf("insert into %s values(1)", tbl))
-		}
 
-		if err != nil {
-			// table t1 executes failed
-			// table t2 executes successfully
-			_, err = exec(s, "commit")
-		} else if tbl == "t2" {
-			err = errors.New("insert result isn't expected")
-		}
+		_, err = exec(s, fmt.Sprintf("insert into %s values(1)", tbl))
```

Comment:
Can't call mustExec here, because this function run in a goroutine, if it panic without send finish signal to the channel, other goroutine will wait forever.